gitpod-io/gitpod
# [stripe] guard resubscribing with another currency

Currently it's not supported to change the currency when re-subscribing. Instead of failing with with non-telling message, see #13201, we should explain the situation and make changing currency a support case for now. 

<img width="576" alt="Screen Shot 2022-10-21 at 12 07 55" src="https://user-images.githubusercontent.com/914497/197171483-fd5f5540-2dff-435b-8342-5830c337ef77.png">

In addition this PR changes the way the customer is fetched. The StripeCustomerId is persisted per AttributionId in Gitpod's DB with the purpose of being able to fetch the customer details from Stripe's Customer endpoint. Searching by metadata burns the API rate in those cases. But not fetching the Customer details from Stripe and just use the DB entry, may result in a divergence. 

## Description


## Related Issue(s)

Fixes #13201

## How to test
Follow the steps from #13201 and see there is a telling message when you're trying to switch currencies.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
